src/segtree.rs

```diff
@@ -68,6 +68,48 @@ where
     }
 }
 
+pub struct BitOrOper<S>(Infallible, PhantomData<fn() -> S>);
+impl<S> Monoid for BitOrOper<S>
+where
+    S: Copy + BitOr<Output = S> + Default,
+{
+    type S = S;
+    fn identity() -> Self::S {
+        S::default()
```

Comment:
I'm still doubting if we should use "Default" in this case, considering the semantics.  (See also my [previous comment](https://github.com/rust-lang-ja/ac-library-rs/pull/104#discussion_r1149274145)).